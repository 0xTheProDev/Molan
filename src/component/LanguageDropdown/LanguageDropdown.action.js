// @flow
import React from 'react';
import { LanguageDropdownOption, onLanguageChangeCallbackParam } from './LanguageDropdown.flow';

export function onLanguageChangeCallback(onLanguageChange: onLanguageChangeCallbackParam) {
  return function onChangeHandler(event: React.SyntheticEvent, data: LanguageDropdownOption) {
    onLanguageChange(data.value);
  }
}
