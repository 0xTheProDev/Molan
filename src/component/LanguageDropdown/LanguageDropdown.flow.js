// @flow
import React from 'react';

export type onLanguageChangeCallbackParam = (language: string) => void;

export type LanguageDropdownProps = {
  DefaultLanguage: string,
  onLanguageChange: onLanguageChangeCallbackParam,
};

export type LanguageDropdownOption = {
  key: string,
  text: string,
  value: string,
};

export type LanguageDropdownInternalProps = {
  LanguageDropdownPlaceholder: string,
  DefaultLanguage: string,
  LanguageOptions: Array<LanguageDropdownOption>,
  onLanguageChange: (event: React.SyntheticEvent, selectedLanguage: LanguageDropdownOption) => void,
};
