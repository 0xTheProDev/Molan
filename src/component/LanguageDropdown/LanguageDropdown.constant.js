// @flow
import { LANGUAGES, COMPILER_OPTIONS } from 'constants/templates';

import { LanguageDropdownOption } from './LanguageDropdown.flow';

export const LANGUAGE_DROPDOWN_OPTIONS: Array<LanguageDropdownOption> =  LANGUAGES.map(language => {
  return {
    key: language,
    value: language,
    text: COMPILER_OPTIONS[language],
  };
});
