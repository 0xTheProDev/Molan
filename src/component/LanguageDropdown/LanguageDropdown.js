// @flow
import React from 'react';
import { Select } from 'semantic-ui-react';
import { noop } from 'util/common';

import { LanguageDropdownInternalProps } from './LanguageDropdown.flow';
import { LANGUAGE_DROPDOWN_OPTIONS } from './LanguageDropdown.constant';

export function LanguageDropdown(props: LanguageDropdownInternalProps) {
  const {
    LanguageDropdownPlaceholder,
    LanguageOptions,
    DefaultLanguage,
    onLanguageChange,
  } = props;

  return (
    <Select
      placeholder={LanguageDropdownPlaceholder}
      options={LanguageOptions}
      value={DefaultLanguage}
      onChange={onLanguageChange}
    />
  );
}

LanguageDropdown.defaultProps = {
  LanguageDropdownPlaceholder: 'Select a language',
  LanguageOptions: LANGUAGE_DROPDOWN_OPTIONS,
  onLanguageChange: noop,
};
