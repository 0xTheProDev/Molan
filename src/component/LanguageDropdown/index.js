// @flow
import React from 'react';

import { LanguageDropdownProps } from './LanguageDropdown.flow';
import { onLanguageChangeCallback } from './LanguageDropdown.action';
import { LanguageDropdown } from './LanguageDropdown';

export default function LanguageDropdownFactory({ DefaultLanguage, onLanguageChange }: LanguageDropdownProps) {
  const onChangeHandler = onLanguageChangeCallback(onLanguageChange);

  return (
    <LanguageDropdown
      DefaultLanguage={DefaultLanguage}
      onLanguageChange={onChangeHandler}
    />
  );
}

export { LanguageDropdownProps, LanguageDropdownFactory as LanguageDropdown };
