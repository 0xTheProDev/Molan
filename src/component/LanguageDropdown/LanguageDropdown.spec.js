import React from 'react';
import { Select } from 'semantic-ui-react';

import { LanguageDropdown } from './LanguageDropdown';
import { LANGUAGE_DROPDOWN_OPTIONS } from './LanguageDropdown.constant';

describe('Component::LanguageDropdown::LanguageDropdown', () => {
  let
    onLanguageChangeStub,
    minProps;

  beforeAll(() => {
    onLanguageChangeStub = jest.fn();

    minProps = {
      DefaultLanguage: LANGUAGE_DROPDOWN_OPTIONS[0].value,
      onLanguageChange: onLanguageChangeStub,
    };
  });

  it('should render with base structure', () => {
    const wrapper = shallow(<LanguageDropdown {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke onLanguageChange callback when an option is selected', () => {
    const wrapper = shallow(<LanguageDropdown {...minProps} />);

    wrapper
      .find(Select)
      .simulate('change');
    
    expect(onLanguageChangeStub).toHaveBeenCalled();
  });
});
