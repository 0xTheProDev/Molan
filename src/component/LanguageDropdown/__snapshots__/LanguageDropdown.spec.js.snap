// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::LanguageDropdown::LanguageDropdown should render with base structure 1`] = `
<Select
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "key": "c",
        "text": "C (GCC 6.3)",
        "value": "c",
      },
      Object {
        "key": "cpp",
        "text": "C++ (G++ 6.3)",
        "value": "cpp",
      },
      Object {
        "key": "java",
        "text": "Java (OpenJDK 0.8)",
        "value": "java",
      },
      Object {
        "key": "python",
        "text": "Python (Python 2.7)",
        "value": "python",
      },
      Object {
        "key": "python3",
        "text": "Python (Python 3.6)",
        "value": "python3",
      },
      Object {
        "key": "mips",
        "text": "MIPS-IV (Simulator 0.4)",
        "value": "mips",
      },
      Object {
        "key": "verilog",
        "text": "Verilog (iVerilog 10.0)",
        "value": "verilog",
      },
    ]
  }
  placeholder="Select a language"
  value="c"
/>
`;
